### ISSUE TITLE
Request: Automatically detect external changes and offer to reload

### ISSUE THREAD

**Issue by adamantoise** [mike-ward/Markdown-Edit#263]
If I make changes to a Markdown file outside of Markdown Edit, it currently does not detect those changes, and I have to manually reload the file.  It would be much nicer if it watched the directory containing the file with a `FileSystemWatcher`, in case the file changed for any reason (external editor, pull from source control, etc.).  `FileSystemWatcher` is already used to watch for changes in the user settings file, template file, and snippets file, but not yet for Markdown source files that are being edited.

If any changes were detected, it should display a message box indicating that the file was modified and asking the user if they want to reload it (and discard any unsaved changes they might have, if any) or ignore it.

E.g. here's the dialog box that Notepad++ displays when it detects external modifications:

![Notepad++ screenshot](https://i.stack.imgur.com/CEW5C.png)

Note that some text editors will edit a file in-place (open, truncate, write, close), while other text editors will write out a temporary file and then rename it on top of the old file, and some others might delete the original before writing the new file; so making sure that all of those cases are handled correctly can be non-trivial.

**Comment by mike-ward** [comment]
You're right about the non-trivial part. Visual Studio does something like this and I've always found it a bit annoying.

I've sometimes thought the editor should quietly reload the file if when no modifications are pending and warn that the file has changed only when saving. Haven't thought about all the edge cases but it seems less intrusive.
